src/AccountSelector.jsx

```diff
@@ -3,26 +3,53 @@ import React, { useState, useEffect } from "react";
 import { Menu, Dropdown, Container, Icon, Image } from "semantic-ui-react";
 
 export default function NodeInfo(props) {
-  const { keyring, setAccountAddress } = props;
-  const [accountSelected, setAccountSelected] = useState("");
+  const { keyring, setAccountAddress, api } = props;
 
-  // get the list of accounts we possess the private key for
+  // Get the list of accounts we possess the private key for
   const keyringOptions = keyring.getPairs().map(account => ({
     key: account.address,
     value: account.address,
     text: account.meta.name.toUpperCase(),
     icon: "user"
   }));
 
-  const initialAddress = keyringOptions.length > 0
+  // Setup state
+  const [accountBalance, setAccountBalance] = useState(0);
+  const [unsubBalance, setUnsubBalance] = useState();
+  const [accountSelected, setAccountSelected] = useState(
+    keyringOptions.length > 0
     ? keyringOptions[0].value
-    : "";
+    : ""
+  );
+
+  const onChange = (address) => {
+    // Update state with new account address
+    setAccountAddress(address);
+    setAccountSelected(address);
+  };
 
-  // Set the initial address
+  // When account address changes, update subscriptions
   useEffect(() => {
-    setAccountSelected(initialAddress);
-    setAccountAddress(initialAddress);
-  }, [setAccountAddress, initialAddress])
+    // Unsubscribe previous account's balance if subscription exists
+    unsubBalance && unsubBalance();
```

Comment:
This it not needed. As soon as a new `useEffect` is called (because one of the dependency changes) the return from the `useEffect` is called (so you are unsubscribed).

Just add a `console.log` in the return of the `useEffect` to see that it's called as soon as you change the selection. Hence you do not need to keep track of the unsubsciptions, and generally do not need the `unsubBalance` state.

Reply:
Ahh, indeed. https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect  Thank you.